src/Atmos/Model/AtmosModel.jl

```diff
@@ -229,12 +228,95 @@ Set the value at the boundary to match the `init_state!` function. This is mainl
+    nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, _...) where {T,M,R,S,BC <: InitStateBC,IS}
   init_state!(m, stateP, auxP, auxP.coord, t)
 end
 
-function init_state!(m::AtmosModel, state::Vars, aux::Vars, coords, t)
-  m.init_state(state, aux, coords, t)
+function init_state!(m::AtmosModel, state::Vars, aux::Vars, coord, t)
+  m.init_state(state, aux, coord, t)
+end
+
+"""
+  DYCOMS_BC <: BoundaryCondition
+  Prescribes boundary conditions for Dynamics of Marine Stratocumulus Case
+"""
+struct DYCOMS_BC <: BoundaryCondition
+end
+function boundarycondition!(bl::AtmosModel{T,M,R,S,BC,IS}, stateP::Vars, diffP::Vars, auxP::Vars,
+    nM, stateM::Vars, diffM::Vars, auxM::Vars, bctype, t, state1::Vars, diff1::Vars, aux1::Vars) where {T,M,R,S,BC <: DYCOMS_BC,IS}
+    # stateM is the 𝐘⁻ state while stateP is the 𝐘⁺ state at an interface. 
+    # at the boundaries the ⁻, minus side states are the interior values
+    # state1 is 𝐘 at the first interior nodes relative to the bottom wall 
+    
+    # Get values from minus-side state
+    ρM = stateM.ρ 
+    UM, VM, WM = stateM.ρu
+    EM = stateM.ρe
+    QTM = stateM.moisture.ρq_tot
+    uM, vM, wM  = UM/ρM, VM/ρM, WM/ρM
+    q_totM = QTM/ρM
+    UnM = nM[1] * UM + nM[2] * VM + nM[3] * WM
+    
+    # Assign reflection wall boundaries (top wall)
+    stateP.ρu = SVector(UM - 2 * nM[1] * UnM, 
+                        VM - 2 * nM[2] * UnM,
+                        WM - 2 * nM[3] * UnM)
+
+    # Assign scalar values at the boundaries 
+    stateP.ρ = ρM
+    stateP.moisture.ρq_tot = QTM
+    # Assign diffusive fluxes at boundaries
+    diffP = diffM
+    xvert = auxM.coord[3]
+    
+    if xvert < 0.00001
+      # ------------------------------------------------------------------------
+      # (<var>_FN) First node values (First interior node from bottom wall)
+      # ------------------------------------------------------------------------
+      z_FN             = aux1.coord[3]
+      ρ_FN             = state1.ρ
+      U_FN, V_FN, W_FN = state1.ρu
+      E_FN             = state1.ρe
+      u_FN, v_FN, w_FN = U_FN/ρ_FN, V_FN/ρ_FN, W_FN/ρ_FN
+      windspeed_FN     = sqrt(u_FN^2 + v_FN^2 + w_FN^2)
+      q_tot_FN         = state1.moisture.ρq_tot / ρ_FN
+      e_int_FN         = E_FN/ρ_FN - 0.5*windspeed_FN^2 - grav*z_FN
+      TS_FN            = PhaseEquil(e_int_FN, q_tot_FN, ρ_FN) 
+      T_FN             = air_temperature(TS_FN)
+      q_vap_FN         = q_tot_FN - PhasePartition(TS_FN).liq
+      # --------------------------
+      # Bottom boundary quantities 
+      # --------------------------
+      zM          = auxM.coord[3] 
+      q_totM      = QTM/ρM
+      windspeed   = sqrt(uM^2 + vM^2 + wM^2)
+      e_intM      = EM/ρM - 0.5*windspeed^2 - grav*zM
+      TSM         = PhaseEquil(e_intM, q_totM, ρM) 
+      q_vapM      = q_totM - PhasePartition(TSM).liq
+      TM          = air_temperature(TSM)
+      # ----------------------------------------------------------
```

Comment:
It looks like more quantities are defined here than we really need.

Reply:
Yes, but as long as @asraero has this code working for me as it is, I can then take care of all the cleanup. The reference driver I passed him has a lot of things that are there for testing purposes only but will disappear after I have made sure that the dycoms in the new code works correctly.